tests/unit/rate_limiting/test_integration.py

```
@@ -0,0 +1,297 @@
+"""
+Integration tests for the complete rate limiting system.
+"""
+
+import pytest
+import asyncio
+from unittest.mock import AsyncMock, patch, MagicMock
```

Ruff (0.11.9)

`unittest.mock.MagicMock` imported but unused

Remove unused import: `unittest.mock.MagicMock`

(F401)